pkg/api/types.go

```diff
@@ -1664,6 +1667,104 @@ type SecretList struct {
 	Items []Secret `json:"items"`
 }
 
+// SecurityContext holds security configuration that will be applied to a container.  If a security context
+// is set on the container spec then it must comply with any constraints defined in the SecurityConstraints context
+// it is running in.  If a security context is not supplied and the pod is running under a SecurityConstraints context
+// then a default SecurityContext may be applied.
+type SecurityContext struct {
+	// Capabilities are the capabilities to add/drop when running the container
+	// TODO: will need to refactor this from the container spec to here
+	Capabilities Capabilities
+
+	// Run the container in privileged mode
+	// TODO: will need to refactor this from the container spec to here
+	Privileged bool
+
+	// SELinuxOptions are the labels to be applied to the container
+	// and volumes
+	SELinuxOptions
```

Comment:
This is, admittedly, pretty specific so I would expect the API would have options that grew as other security facilities are supported much like how persistent disks must be implementation specific (https://github.com/markturansky/kubernetes/blob/pv_api/pkg/api/types.go#L177).  After a quick discussion with @pmorie I decided to just go with it as is but if there is a good way to make some of this more generic I'll change it.

In particular, there is already an app armor profile flag supported by docker so we could already have a second `AppArmorOptions`.